### PR TITLE
[BOLT] Add itrace aggregation for AUX data

### PR DESCRIPTION
**Pull request by jonathandavies-arm**
If you have a perf.data with Arm ETM data the only way to use perf2bolt with Branch Aggregation is to first run `perf inject --itrace=l64i1us -o perf-brstack.data` and then pass the new perf-brstack.data into perf2bolt. perf2bolt then runs `perf script -F pid,ip,brstack` to produce the brstacks.

This PR adds `--itrace` arg to perf2bolt to enable Itrace Aggregation. It takes a string which is what is passed to the `perf script -F pid,ip,brstack --itrace={0}`. This command produces the brstacks without having to run perf inject and creating a new perf.data file.

### CODE REVIEW

**Comment by jonathandavies-arm** [comment]
Question for reviewers: Do any Bolt tests run Bolt and pass in a perf.data file and check the .fdata is correct?

**Comment by jonathandavies-arm** [comment]
@maksfb @rafaelauler Please can I have a review.

**Comment by yota9** [comment]
Please use [BOLT] in commit message

**Comment by maksfb** [comment]
> Question for reviewers: Do any Bolt tests run Bolt and pass in a perf.data file and check the .fdata is correct?

We have several internal tests for `perf2bolt`, but nothing in open source. @aaupov, is there a plan to port those?

File: bolt/lib/Profile/DataAggregator.cpp [hunk 1]
```diff
@@ -155,6 +160,8 @@ void DataAggregator::findPerfExecutable() {
 }
 
 void DataAggregator::start() {
+  std::string ItracePerfScriptArgs;
```
Comment (maksfb) [inline bolt/lib/Profile/DataAggregator.cpp:163]: Is there a reason to declare `ItracePerfScriptArgs` in this scope?

**Review by maksfb — COMMENTED** [review]
Thanks! Overall looks good to me, but please check the question about the scope.

File: bolt/lib/Profile/DataAggregator.cpp (hunk 1 above)
Comment (jonathandavies-arm) [reply bolt/lib/Profile/DataAggregator.cpp]: I just saw other std::string being declared at the beginning of functions. I've moved the declaration into the if statement scope.

**Review by maksfb — APPROVED** [review]

**Comment by aaupov** [comment]
> > Question for reviewers: Do any Bolt tests run Bolt and pass in a perf.data file and check the .fdata is correct?
> 
> 
> 
> We have several internal tests for `perf2bolt`, but nothing in open source. @aaupov, is there a plan to port those?

Noted. I didn't port those as we had to sanitize inputs and perf data is tricky since it's opaque and captures a lot of system information by default. We do have some perf2bolt tests in rafaelauler/bolt-tests repo, but not a comprehensive suite.

CC @lanza

**Pull request merged**